src/debug/debug.cc

```diff
@@ -4265,13 +4261,12 @@ GetOrCreatePromiseDependencyGraphData(Isolate* isolate, Handle<Object> promise)
   return iter->second;
 }
 
-void AddPromiseDependencyGraphNesting(Isolate* isolate, Handle<Object> promise, Handle<Object> nested) {
+void AddPromiseDependencyGraphAdoption(Isolate* isolate, Handle<Object> promise, Handle<Object> adopted) {
   PromiseDependencyGraphData& data = GetOrCreatePromiseDependencyGraphData(isolate, promise);
-  v8::Isolate* v8_isolate = (v8::Isolate*) isolate;
-  data.nested_persistent_id = RecordReplayObjectId(v8_isolate,
-                                                   v8_isolate->GetCurrentContext(),
-                                                   v8::Utils::ToLocal(nested),
-                                                   /* allow_create */ true);
+  PromiseDependencyGraphData& adopted_data = GetOrCreatePromiseDependencyGraphData(isolate, adopted);
+
+  recordreplay::AddDependencyGraphEdge(data.new_node_id, adopted_data.new_node_id,
```

Comment:
why "adoptedPromise"? this is (roughly) how Promise A+ spec refers to this process: https://promisesaplus.com/#the-promise-resolution-procedure